Source/WebCore/rendering/RenderElement.cpp

```diff
@@ -1719,12 +1722,14 @@ void RenderElement::imageContentChanged(CachedImage& cachedImage)
         if (cachedImage.hasPaintedHDRContent())
             document().setHasPaintedHDRContent();
     }
+
+    if (document().hasPaintedHDRContent()) {
+        if (auto layer = enclosingCompsitedLayer())
```

Comment:
I think it's better to keep using `enclosingLayer()`, and have the RenderLayer code find the enclosing compositing layer if necessary. RenderElement shouldn't need to think about compositing.

Reply:
Can you please explain a little bit what is needed here? Do you mean adding a flag to `enclosingLayer(bool composited)`?